sink/sqs.go

```diff
@@ -146,12 +146,16 @@ func (s *SQSSink) write() {
 
 			err := s.sendBatch(entries)
 			if err != nil && strings.Contains(err.Error(), "AWS.SimpleQueueService.BatchRequestTooLong") {
-				for i := 0; i < len(entries); i += i {
```

Comment:
for anyone following along, the issue with the retry code was `i += i` part of the for loop, since `i := 0` the logic was basically infinite retry with the same message over and over when hitting retries